cli/file_fetcher.rs

```diff
@@ -381,20 +381,23 @@ impl FileFetcher {
       ));
     }
 
-    let blob_url_storage = self.blob_url_store.borrow();
-    let blob = blob_url_storage.get(specifier.clone())?.ok_or_else(|| {
-      custom_error(
-        "NotFound",
-        format!("Blob URL not found: \"{}\".", specifier),
-      )
-    })?;
+    let blob_store = self.blob_store.borrow();
+    let blob =
+      blob_store
+        .get_object_url(specifier.clone())?
+        .ok_or_else(|| {
+          custom_error(
+            "NotFound",
+            format!("Blob URL not found: \"{}\".", specifier),
+          )
+        })?;
 
-    let content_type = blob.media_type;
+    let content_type = blob.media_type.clone();
+    let bytes = blob.read_all().await?;
```

Comment:
Looks like borrow to `blob_store` is held across this await point - if there are two blobs read concurrently it will probably panic.